src/plugins/recordTypes/collectionobject/fields.js

```diff
@@ -47,6 +47,16 @@ export default (configContext) => {
             },
           },
         },
+        objectCountGroupList: {
+          objectCountGroup: {
+            [config]: {
+              repeating: false,
```

Comment:
This is the thing that causes search to break. In order to generate the right search query, we need to know that `objectCountGroup` is repeating, so this needs to stay true. The trick is to make it not repeating only in the form.